### DISCUSSION TITLE
How to use custom spacing with extend?

### DISCUSSION THREAD

**Discussion by mustafa-soylemez** [tailwindlabs/tailwindcss#12966] [Help] [answered]
I changed default spacing values to some custom values like 

```
theme: {
       spacing: {
            5.5: '1.375rem',
       }
}
```

It works perfectly with width, padding and other classes which uses spacing values by default. But when i add spacing values to a class, custom classes is not working

```
theme: {
       extend: {
              fontSize: { ...theme.spacing },
       }
}
```

What am i doing wrong?

**Comment by alex-krasikau** [comment, marked as answer]
Hey @mustafa-soylemez!

To use extended spacing in the ﻿`fontSize`, you need to use Tailwind CSS's ﻿theme functions as they allow us to reference the values from the ﻿theme object.

Here's how you can do it:
```js
module.exports = {
    theme: {
        extend: {
            fontSize: (theme) => ({
                'extra': theme('spacing[5.5]'),
                // more sizes...
            }),
            spacing: {
                '5.5': '1.375rem',
                // other spacing...
            }
        }
    }
}
```
In this code:
	1.	﻿`(theme) => ({...})` allows you to access the theme object inside the ﻿extend section.
	2.	﻿`theme('spacing[5.5]')` allows you to reference the newly-defined spacing value.
	3.	`﻿'extra': theme('spacing[5.5]')` adds a new class ﻿text-extra that applies your ﻿1.375rem spacing as font size.

After doing this, you can use ﻿`.text-extra` class in your HTML and it will apply `﻿font-size: 1.375rem`.

**Reply by mustafa-soylemez** [reply]
Hello is there a way to define all custom spacing to fontSize without listing all of them one by one?

**Reply by mustafa-soylemez** [reply]
solved by editing your code thanks for the lead 
`fontSize: (theme) => ({ ...theme('spacing') }),`

**Reply by alex-krasikau** [reply]
Here's how you can do it. Although I gotta say, it's not recommended to handle your font sizes like this. It's better to stick to the scale that has a bunch of pre-set values, even if they're different from the default config in Tailwind CSS.

```js
export default {
  theme: {
    extend: {
      fontSize: (theme) => ({
        ...theme('spacing'),
      }),
      spacing: {
        5.5: '1.375rem',
      },
    },
  },
  plugins: [],
}

```
